### PR TITLE
mannualy specify url for boost

### LINKED ISSUE #2282: build requirement fails on linux
error downloading:

        --- LOG END ---
         error: downloading 'https://dl.bintray.com/boostorg/release/1.66.0/source/boost_1_66_0.7z' failed
         status_code: 22
         status_string: "HTTP response code said error"
         log:

### PR DESCRIPTION
**Pull request by zt-chen**
The sunset of bintray invalidated the url of boost source (boostorg/boost#502), and the https://github.com/ruslo/hunter repo is not being maintained anymore. 

So one solution is to manually specify the new URL of the boost source archive, or  https://github.com/ethereum-mining/ethminer/blob/ce52c74021b6fbaaddea3c3c52f64f24e39ea3e9/CMakeLists.txt#L17 need to be updated to use an actively maintained hunter. 

This is a small patch aim to fix #2282

### CODE REVIEW

**Comment by zt-chen** [comment]
This should be able to fix #2290 too

**Review by xaliander — APPROVED** [review]
**Approved change:**
The error described below is solved by this commit. However, the URL could also be updated to the newest boost version. Please review @aloisklink.

**Root of the problem:**
Boost has moved downloads to JFrog Artifactory [[1](https://www.boost.org/users/news/boost_has_moved_downloads_to_jfr.html)].

**Error message:**
```
-- File already exists but hash mismatch. Removing...
-- Downloading...
   dst='/home/user/.hunter/_Base/Download/Boost/1.66.0/075d0b4/boost_1_66_0.7z'
   timeout='none'
-- Using src='https://dl.bintray.com/boostorg/release/1.66.0/source/boost_1_66_0.7z'
-- Retrying...
-- Using src='https://dl.bintray.com/boostorg/release/1.66.0/source/boost_1_66_0.7z'
-- Retry after 5 seconds (attempt #2) ...
-- Using src='https://dl.bintray.com/boostorg/release/1.66.0/source/boost_1_66_0.7z'
-- Retry after 5 seconds (attempt #3) ...
-- Using src='https://dl.bintray.com/boostorg/release/1.66.0/source/boost_1_66_0.7z'
-- Retry after 15 seconds (attempt #4) ...
-- Using src='https://dl.bintray.com/boostorg/release/1.66.0/source/boost_1_66_0.7z'
-- Retry after 60 seconds (attempt #5) ...
-- Using src='https://dl.bintray.com/boostorg/release/1.66.0/source/boost_1_66_0.7z'
CMake Error at Build/Boost-prefix/src/Boost-stamp/download-Boost.cmake:159 (message):
  Each download failed!
```

File: cmake/Hunter/config.cmake
Comment (aloisklink) [inline cmake/Hunter/config.cmake:7]: A better way of doing this would be to upgrading hunter to the latest version, by changing lines https://github.com/ethereum-mining/ethminer/blob/ce52c74021b6fbaaddea3c3c52f64f24e39ea3e9/CMakeLists.txt#L16-L20 to:

```cmake
HunterGate(
    URL "https://github.com/cpp-pm/hunter/archive/v0.23.320.tar.gz"
    SHA1 "9b4e732afd22f40482c11ad6342f7d336634226f"
    LOCAL
)
```

and specifying a version for `jsoncpp` (the latest version by hunter changes linking).

```suggestion
hunter_config(Boost VERSION 1.66.0)
hunter_config(jsoncpp VERSION 1.8.0)
```

But this also works!

**Review by aloisklink — APPROVED** [review]
It looks good to me, but it would be slightly better to instead update hunter.

@arrichter, I'm not a maintainer of this repo, so I unfortunately can't merge this :cry:
I don't think there's anyone with admin access actively maintaining this repo, so it's unlikely for this to get merged.
It might be worth contacting one of the people in https://github.com/ethereum-mining/ethminer#maintainers--authors if you want this to be merged (and maybe give them a tip if you do, since they're not getting paid for open-source).

@ChazyTheBest, you should be able to use the system boost library by setting [`HUNTER_ENABLED=NO`](https://hunter.readthedocs.io/en/latest/reference/user-variables.html?highlight=HUNTER_ENABLED#hunter-enabled), e.g. by doing `cmake .. -DHUNTER_ENABLED=NO`

**Pull request closed without merging**
